File: MatlabOctaveCodeSnippets/snip_signals_sinusoid_generation.m
Fs=8000; %sampling frequency (Hz)
Ts=1/Fs; %sampling interval (seconds)
N=20000; %number of desired samples
f0=600; %cosine frequency (Hz)
%%%%First alternative to generate a cosine of 600 Hz
t=0:Ts:(N-1)*Ts; %discretized continuous-time axis (sec.)
x1=5*cos(2*pi*f0*t); %amplitude=5 V and frequency = f0 Hz
%%%%Second alternative: work directly in discrete-time
w0=2*pi*f0*Ts; %w0 is in rad, convert from rad/s to rad
n=0:N-1; %discrete-time axis (do not use Ts anymore)
x2=5*cos(w0*n); %amplitude=5 V and frequency = w0 rad
plot(x1-x2); %plot error between two alternative sequences
soundsc(x1,Fs) %for fun: playback one of them to listen

File: PythonCodeSnippets/snip_signals_sinusoid_generation.py
import matplotlib.pyplot as plt
import numpy as np
import pylab as pl
from scipy.io import wavfile

Fs = 8000  # Defining a sampling frequency (Hz)
Ts = 1 / Fs  # Defining a sampling period (seconds)
N = 20000  # Number of samples
f0 = 600  # Cosine frequency in Hz
# First method to generate a cosine with frequency equals to 600 Hz
t = np.arange(0, (N - 1) * Ts, Ts)  # Discretized continuous-time (seconds)
x1 = 500 * np.cos(2 * np.pi * f0 * t)
# Second metod: work with in discrete-time
w0 = 2 * np.pi * f0 * Ts  # w0 is the discrete-time angular frequency (radians)
n = np.arange(0, N - 1, 1)  # Using a discrete-time axis
x2 = 500 * np.cos(w0 * n)  # Getting that cosine now with this new method
# Getting the error between that two metods
Error = x1 - x2
print("Average error =", np.mean(Error))
plt.plot(n, Error)
pl.show()
wavfile.write("output.wav", Fs, x1.astype(np.int16))  # save the first signal to a wav file. 

#Obs: If you don't hear the audio try increasing (x100) the cosine amplitude


